homeassistant/components/ffmpeg_motion/binary_sensor.py

```diff
@@ -55,15 +56,16 @@ async def async_setup_platform(hass, config, async_add_entities, discovery_info=
 
 
 class FFmpegBinarySensor(FFmpegBase, BinarySensorEntity):
-    """A binary sensor which use FFmpeg for noise detection."""
+    """A binary sensor which use FFmpeg for detection."""
 
     def __init__(self, config):
-        """Init for the binary sensor noise detection."""
+        """Init for the ffmpeg binary sensor detection."""
         super().__init__(config.get(CONF_INITIAL_STATE))
 
         self._state = False
         self._config = config
-        self._name = config.get(CONF_NAME)
+        self._attr_name = config.get(CONF_NAME)
+        self._attr_unique_id = config.get(CONF_UNIQUE_ID)
```

Comment:
What happens if this isn't provided? The entity has no unique ID?

Reply:
Just like a template entity the [unquie_id](https://www.home-assistant.io/integrations/switch.template/#unique_id) is optional. Adding a unique ID allows customizing the entity from the front end.